core/src/main/scala/org/broadinstitute/dsde/rawls/entities/base/ExpressionEvaluationSupport.scala

```diff
@@ -24,9 +24,42 @@ trait ExpressionEvaluationSupport {
     valuesByEntity.map({ case (entityName, values) => SubmissionValidationEntityInputs(entityName, values.toSet) }).toStream
   }
 
+  protected def isStringInputType(input: MethodInput): Boolean = {
+    val valuetype = input.workflowInput.getValueType
+
+    // TODO: map types? tuple types?
+    valuetype.getTypeName == TypeNameEnum.STRING ||
+      (valuetype.getTypeName == TypeNameEnum.OPTIONAL &&
+        valuetype.getOptionalType.getTypeName == TypeNameEnum.STRING) ||
+      (valuetype.getTypeName == TypeNameEnum.ARRAY &&
+        valuetype.getArrayType.getTypeName == TypeNameEnum.STRING)
+  }
+
+  protected def maybeConvertToString(rawValue: Attribute): Attribute = {
+    // cast numbers and booleans to strings; leave other types alone
+    rawValue match {
+      case n:AttributeNumber => AttributeString(n.value.toString) // n.value.bigDecimal.toPlainString to avoid scientific notation
+      case b:AttributeBoolean => AttributeString(b.value.toString)
+      case l:AttributeValueList =>
+        val convertedValues = l.list.map(maybeConvertToString) collect {
+          case av:AttributeValue => av
+        }
+        AttributeValueList(convertedValues)
+      case _ => rawValue
+    }
+  }
```

Comment:
reviewer: what do you think about scientific notation? For very large or very small `AttributeNumber` values, the `toString` will generate scientific notation. Should we leave that as-is since it is the Java default? Or should we use `toPlainString` even if that results in e.g. many many zeros after a decimal point?

Reply:
I think we want `toPlainString` in order to preserve the number. I'm imagining that if a user uploading a very long number as a TSV (in the old/current Terra), we'd import it as a string and pass along a very long string to the workflow. So I think we'd break the workflow if we introduced scientific notation when casting

Reply:
Thinking more about this, Batch may be able to parse the scientific notation to a number just as well. But I'm not confident about this, so vote is to keep things the same (toPlainString)